ci_scripts/utils/git_helpers.py

```diff
@@ -1,163 +1,415 @@
-        subprocess.check_call(
-            ['git', 'config', '--global', 'user.email', 'monty-bot@arm.com'])
-        self.set_remote_url(self.git_url_ssh_to_https(self.get_remote_url()))
+"""Utility script to abstract git operations for our CI scripts."""
+import logging
+import os
+import re
+from git import Repo, Actor
+
+from .definitions import ENVVAR_GIT_TOKEN, REMOTE_ALIAS, PROJECT_ROOT, \
+    MASTER_BRANCH, BETA_BRANCH, RELEASE_BRANCH_PATTERN
+
+logger = logging.getLogger(__name__)
+
+
+class GitWrapper:
+    """Wrapper class to provide convenient methods for performing git actions."""
+
+    def __init__(self):
+        """Creates an instance of GitWrapper."""
+        self.repo = Repo(PROJECT_ROOT)
+        self.author = Actor("monty bot", "monty-bot@arm.com")
+
+    def _git_url_ssh_to_https(self, url: str) -> str:
+        """Changes repository URL to use authorisation token.
+
+        Converts the git url to use the GitHub token:
+        See https://github.blog/2012-09-21-easier-builds-and-deployments-using-git-over-https-and-oauth/
+
+        Returns:
+            new URL
+        """
+        path = url.split('github.com', 1)[1][1:].strip()
+        new = 'https://{GITHUB_TOKEN}:x-oauth-basic@github.com/%s' % path
+        logger.info('rewriting git url to: %s' % new)
+        return new.format(GITHUB_TOKEN=os.getenv(ENVVAR_GIT_TOKEN))
+
+    def configure_author(self):
+        """Sets the author."""
+        self.repo.config_writer().set_value("user", "name",
+                                            self.author.name).release()
+        self.repo.config_writer().set_value("user", "email",
+                                            self.author.email).release()
+
+    def checkout(self, branch) -> None:
+        """Checks out a branch.
+
+        Args:
+            branch: branch to check out
+
+        """
+        self.repo.git.checkout(branch)
+
+    def _add_one_file(self, path: str) -> None:
+        relative_path = os.path.relpath(path, start=PROJECT_ROOT)
+        if relative_path:
+            relative_path = relative_path.replace('\\', '/')
+            if os.path.isdir(relative_path):
+                relative_path = f'{relative_path}/*'
+        self.repo.git.add(relative_path)
+
+    def add(self, path) -> None:
+        """Adds a file or a list of files.
+
+        Args:
+            path: file path or list of file paths
+        """
+        if isinstance(path, list) or isinstance(path, set):
+            for element in path:
+                self.add(element)
+        else:
+            self._add_one_file(path)
+
+    def commit(self, message: str, **kwargs) -> None:
+        """Commits changes to the repository.
+
+        Args:
+            message: commit message
+            **kwargs: extra parameters
+        """
+        self.repo.index.commit(
+            message,
+            author=self.author,
+            **kwargs
+        )
+
+    def get_master_branch(self):
```

Comment:
No return type.

Reply:
Same question here - is this not enforced by the linter?